### ISSUE TITLE
Issues with history

### ISSUE THREAD

**Issue by modem7** [sabnzbd/sabnzbd#2767]
### SABnzbd version

4.2.0

### Operating system

Docker

### Using Docker image

linuxserver

### Description

Since upgrading to 4.2.0 items go into history until they complete, and then immediately remove themselves, which breaks integration with other tools. 

Downgrading to 4.1.0 resolves. 

There are no errors or messages in the console unfortunately that would indicate an error, download completes successfully.

```yaml
 #SABnzbd - Binary newsgrabber (NZB downloader)
  sabnzbd:
    # image: linuxserver/sabnzbd
    image: linuxserver/sabnzbd:4.1.0
    container_name: Sabnzbd
    hostname: Sabnzbd
    environment:
      PUID: $PUID
      PGID: $PGID
      TZ: $TZ
      DOCKER_MODS: linuxserver/mods:sabnzbd-par2cmdline-turbo
      # DOCKER_MODS: ghcr.io/gilbn/theme.park:sabnzbd
      # TP_THEME: Overseer
    logging:
      driver: "local"
      options:
        max-size: 10m
        max-file: "3"
    labels:
      # - backup
      - autoheal=true
      - "traefik.enable=true"
      ## HTTP Routers Auth Bypass
      - "traefik.http.routers.sabnzbd-rtr-bypass.entrypoints=https"
      - "traefik.http.routers.sabnzbd-rtr-bypass.rule=Host(`sabnzbd.$DOMAINNAME`) && Query(`apikey`, `$SABNZBD_API_KEY`)"
      - "traefik.http.routers.sabnzbd-rtr-bypass.priority=100"
      # - "traefik.http.routers.sabnzbd-rtr-bypass.tls=true"
      ## HTTP Routers Auth
      - "traefik.http.routers.sabnzbd-rtr.entrypoints=https"
      - "traefik.http.routers.sabnzbd-rtr.rule=Host(`sabnzbd.$DOMAINNAME`)"
      - "traefik.http.routers.sabnzbd-rtr.priority=98"
      # - "traefik.http.routers.sabnzbd-rtr.tls=true"
      ## Middlewares
      - "traefik.http.routers.sabnzbd-rtr-bypass.middlewares=chain-no-auth@file"
      # - "traefik.http.routers.sabnzbd-rtr-ping.middlewares=chain-no-auth@file"
      - "traefik.http.routers.sabnzbd-rtr.middlewares=chain-oauth-admins@file"
      # - "traefik.http.routers.sabnzbd-rtr.middlewares=chain-authelia@file" # Authelia
      ## HTTP Services
      - "traefik.http.routers.sabnzbd-rtr.service=sabnzbd-svc"
      - "traefik.http.routers.sabnzbd-rtr-bypass.service=sabnzbd-svc"
      # - "traefik.http.routers.sabnzbd-rtr-ping.service=sabnzbd-svc"
      - "traefik.http.services.sabnzbd-svc.loadbalancer.server.port=8080"
      ## Homepage
      - homepage.name=Sabnzbd
      - homepage.group=Download + Search
      - homepage.icon=sabnzbd
      - homepage.href=https://sabnzbd.$DOMAINNAME
      - homepage.widget.type=sabnzbd
      - homepage.widget.url=http://sabnzbd:8080
      - homepage.widget.key=$SABNZBD_API_KEY
    networks:
      pihole:
        ipv4_address: '172.22.0.103'
    healthcheck:
      test: curl -fSs 127.0.0.1:8080/api?mode=version || exit 1
      interval: 5s
      timeout: 10s
      retries: 3
    security_opt:
      - no-new-privileges:true
    ports:
      - "25916:8080"
    volumes:
      - $USERDIR/Sabnzbd:/config
      - $DOWNLOADDIR:/downloads
      - $SABDIR:/incomplete-downloads
      # - $DOWNLOADDIR:/incomplete-downloads
      # - $RAMDRIVEPLEX/sab:/incomplete-downloads
    restart: always
```

**Comment by Safihre** [comment]
Could you do the following:
Enable Debug logging in the Status window.
Make sure all downloads are paused or the queue is empty.
Restart SABnzbd (4.2.0).
Download the 100MB test download from the Status window.
Does it appear in the history?
If not, click Show Logging in the Status window and share the whole log after the restart, including the anonymous copy of the settings at the end. For example through pastebin.

**Comment by modem7** [comment]
Typical, the moment I need to replicate the issue is starts behaving. 

Cheers for the quick response! I'll consider this closed for now, no idea why the behaviour occurred nor why it resolved itself (prior to posting I had fully removed the container + image + redownloaded). 

If it re-occurs, I'll reopen the ticket w/ the information required above. 

Thanks again!

**Comment by Safihre** [comment]
Could it be that those jobs were already in the queue or waiting for post processing before the update to 4.1.0? Maybe I missed something with the upgrade.

**Comment by mnightingale** [comment]
> ... which breaks integration with other tools

If *arr then check download client settings, there is "remove imported dwonloads from download client history" option near the bottom.

**Comment by modem7** [comment]
> Could it be that those jobs were already in the queue or waiting for post processing before the update to 4.1.0? Maybe I missed something with the upgrade.

Narp, checked the queue + history was totally clean and empty prior in this case.



> > ... which breaks integration with other tools
> 
> If *arr then check download client settings, there is "remove imported dwonloads from download client history" option near the bottom.

One of the first things I checked. Thought it was human error (as it usually is let's be honest), but for now, it seems that it was some odd docker behaviour.

I'll keep monitoring to see if it's resolved. Given the image was released over 19 hours ago, I'd have expected more people to comment if this were a larger scale issue, but I'm fairly confident it was a local problem for now.

**Comment by imokaythanks** [comment]
I have gotten this issue after upgrading to 4.2.0 so it's not just you. I will see if I can figure anything out from logs when I get a chance

**Comment by imokaythanks** [comment]
Mine also appears to have resolved itself. Could not reproduce again.

Only unique info I have is that the download failed because of missing articles
```
2024-01-03 14:14:50,841::INFO::[nzbstuff:1804] [N/A] Purging data for job redacted (delete_all_data=False)
2024-01-03 14:14:50,842::INFO::[notifier:137] Sending notification: Download Failed - redacted (type=failed, job_cat=movies)
2024-01-03 14:15:18,718::INFO::[database:278] Added job redacted to history
2024-01-03 14:15:18,720::INFO::[postproc:137] Saving postproc queue
2024-01-03 14:15:18,721::INFO::[downloader:416] Post-processing finished, resuming download
2024-01-03 14:15:22,722::INFO::[nzbqueue:230] Saving queue
2024-01-03 14:15:22,725::INFO::[postproc:137] Saving postproc queue
2024-01-03 14:15:22,726::INFO::[notifier:137] Sending notification: SABnzbd - Queue finished (type=queue_done, job_cat=None)
2024-01-03 14:15:22,726::INFO::[postproc:1102] Queue has finished, launching script: None
```

Hours later I noticed that radarr never got notified about the failed download and it was not in sabnzbd history. I manually queued up more downloads that eventually failed for missing articles and watched them disappear from history.

Today I restarted the container, did a successful test download, and redid yesterdays process and it all had the expected automation

**Comment by Safihre** [comment]
@imokaythanks which Docker image do you use?

**Comment by thezoggy** [comment]
had a user report the same issue in discord, they were using 4.1.0 and had alpine 3.18 from docker maintainer. last week container got alpine 3.19 with a lot of packages.. so when they updated their container today to get 4.2.0 they also got that upgrade. but dont know if how they upgrade causes anything to have stale stuff or gremlins from anything. nothing stood out in logs about permission issues/failures. having them restart their docker everything just worked again while on 4.2.0

**Comment by mnightingale** [comment]
The discussion has reminded me that at some point I did run into the above where the database was locked, restarting resolved it and since I couldn't recreate it I couldn't be sure it was related to 4.2.0
I also run linuxserver unstable and have been running various 4.2.0 releases.

**Comment by Safihre** [comment]
My current assessment is that we don't explicitly 'commit' the change of database column names in the connect() call. Because of that the insert fails. 
I'm a bit surprised the history insert fails silently..

**Comment by paulmorabito** [comment]
I'm having the same problem too. Using the linuxserver container and this only started occurring with v4.2. LMK if I can help with debugging.

**Comment by Safihre** [comment]
I found the problem, thank you all!
From the log provided by @thezoggy it shows that indeed the changes to the database structure were not commited, leaving the database in a broken state.
Additionally, the error handling was wrong so it would fail silently.
Will release a 4.2.1 with the fix.

**Comment by Safihre** [comment]
Closes by 0db5ae83901b98142001b32c0794c1538c35d80a
